L1Trigger/L1TMuonEndCap/src/PatternRecognition.cc

```diff
@@ -287,6 +287,8 @@ bool PatternRecognition::is_zone_empty(
     EMTFHitCollection::const_iterator conv_hits_end = ext_conv_hits_it->end();
 
     for (; conv_hits_it != conv_hits_end; ++conv_hits_it) {
```

Comment:
I can let @jiafulow answer perhaps ... but what exactly is the advantage of a "range based loop" over an implementation like this?  Performance?  Style?